pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -183,7 +183,7 @@ func certClass(ctx context.Context) string {
 }
 
 func (r *Reconciler) tls(ctx context.Context, dm *v1alpha1.DomainMapping) ([]netv1alpha1.IngressTLS, []netv1alpha1.HTTP01Challenge, error) {
-	if !autoTLSEnabled(ctx, dm) {
+	if !autoTLSEnabled(ctx, dm) && dm.Spec.TLS == nil {
```

Comment:
I'd actually land this line in a follow-on PR I think: we want to think through the cases here (eg if autotls is not enabled and spec.TLS is non-nil but SecretName is empty what should happen? Should we error out if the cert doesn't exist but autotls is disabled?) and probably add some test coverage.